### PR TITLE
Adds scrolling to webchat

### PR DESCRIPTION
**Pull request by shefalinayak**
## Page to edit
url: https://dev.va.gov/coronavirus-chatbot/

## Origin of request (internal/stakeholder/user feedback)
Related to this [issue](https://github.com/department-of-veterans-affairs/covid19-chatbot/issues/128)

## Description of what's needed (edits/link changes/additions)

- When the user clicks on a button, the page scrolls to the last message from the user (which is the first message of the new content)
- Scrolling is smooth (vs jumping to new location)

### CODE REVIEW

File: src/applications/coronavirus-chatbot/utils.js [hunk 1]
```diff
@@ -7,6 +15,8 @@ const disableButtons = event => {
   for (let i = 0; i < siblingButtons.length; i++) {
     siblingButtons[i].disabled = true;
   }
+
+  scrollToNewMessage();
```
Comment (cvalarida) [inline src/applications/coronavirus-chatbot/utils.js:19]: I'd recommend either
1. Adding a separate event listener for this after line 26, or
2. Creating a new `questionAnswered` function which calls both `disableButtons` and `scrollToNewMessage`

File: src/applications/coronavirus-chatbot/index.js [hunk 2]
```diff
@@ -153,6 +153,6 @@ const chatRequested = scenario => {
 
 export default function initializeChatbot(_root) {
   root = _root;
-  watchForButtonClicks();
+  watchForButtonClicks(root);
```
Comment (cvalarida) [inline src/applications/coronavirus-chatbot/index.js:156]: Looks like this is vestigial from testing? `watchForButtonClicks` doesn't take an argument.

**Review by cvalarida — COMMENTED** [review]
What's the story behind the changes to `yarn.lock`? I see nothing removed from `package.json` that explains it, and I thought @katieho-va already fixed up the lock file after removing `botframework-webchat`. :confused:

File: src/applications/coronavirus-chatbot/utils.js (hunk 1 above)
Comment (katieho-va) [reply src/applications/coronavirus-chatbot/utils.js:19]: We're implementing option 2!

File: src/applications/coronavirus-chatbot/index.js (hunk 2 above)
Comment (katieho-va) [reply src/applications/coronavirus-chatbot/index.js:156]: Yes, will remove root

**Comment by katieho-va** [comment]
@cvalarida Just resolved the differences in `yarn.lock`, the changes shouldn't have been in this PR

**Review by cvalarida — APPROVED** [review]
Beautiful! :shipit:

File: src/applications/coronavirus-chatbot/utils.js [hunk 3]
```diff
@@ -14,6 +22,7 @@ export const watchForButtonClicks = () => {
     const buttons = document.getElementsByClassName('ac-pushButton');
     for (let i = 0; i < buttons.length; i++) {
       buttons[i].addEventListener('click', disableButtons);
+      buttons[i].addEventListener('click', scrollToNewMessage);
```
Comment (ncksllvn) [inline src/applications/coronavirus-chatbot/utils.js:25]: This is confusing - two event listeners applied to each element in a dynamic list of elements...I'm wondering if we can reduce this to -

```js
// When the webchat is starting up - this is attached to the root element and only attached once
root.addEventListener('click', event => {
    const node = event.target;
    const isButton = node.classlist.includes('ac-pushButton');
    if (!isButton) return;
    
    disableButtons();
    scrollToNewMessage();
});
```

Comment (shefalinayak) [reply src/applications/coronavirus-chatbot/utils.js:25]: This PR was merged quickly, but we looked into this approach afterwards:
The event listener on each button (added by botframework-webchat) doesn't bubble the event up so our event listener on root was never getting executed. It seems messier to change the webchat code, so we stuck with applying our event listener dynamically.

**Pull request merged**